apps/dialtone-documentation/docs/.vuepress/theme/components/PageToc.vue

```diff
@@ -1,29 +1,103 @@
       On this page
     </h2>
-    <toc
-      :headers="$page.headers"
-      :options="options"
-    />
+    <nav>
+      <ul
+        v-for="header in internalHeaders"
+        :key="header.slug"
+      >
+        <li v-if="!header.children.length">
+          <toc-item
+            class="d-fw-semibold"
+            :active="isItemActive(header)"
+            :to="header.link"
+            :text="header.title"
+          />
+        </li>
+        <dt-collapsible
+          v-else
+          element-type="li"
+          class="lg:d-d-flex d-fw-wrap"
+          :open="isHeaderActive(header)"
+        >
+          <template #anchor="{ attrs }">
+            <toc-item
+              v-bind="attrs"
+              class="d-fw-semibold"
+              :active="isItemActive(header)"
+              :to="header.link"
+              :text="header.title"
+            />
+          </template>
+          <template
+            v-if="header.children.length"
+            #content
+          >
+            <ul class="d-pl8">
+              <li
+                v-for="child in header.children"
+                :key="child.slug"
+                class="lg:d-d-flex d-fw-wrap"
+              >
+                <toc-item
+                  :active="isItemActive(child)"
+                  class="d-fw-medium"
+                  :to="child.link"
+                  :text="child.title"
+                />
+              </li>
+            </ul>
+          </template>
+        </dt-collapsible>
+      </ul>
+    </nav>
   </aside>
 </template>
 
 <script setup>
-const options = {
-  listClass: 'toc-list d-ls-reset',
-  itemClass: 'toc-item lg:d-d-flex d-fw-wrap',
-  linkTag: 'a',
-  linkClass: 'd-btn d-btn--sm d-btn--muted d-fw-normal d-w100p d-jc-flex-start',
-  linkActiveClass: 'd-btn--active d-fc-primary',
-  linkChildrenActiveClass: '',
-};
+import { ref, watch } from 'vue';
+import { usePageData } from '@vuepress/client';
+import { useRoute } from 'vue-router';
+import TocItem from './TocItem.vue';
+
+const props = defineProps({
+  headers: {
+    type: Array,
+    default: undefined,
+  },
+});
+
+const route = useRoute();
+const internalHeaders = ref(props.headers);
+
+function isHeaderActive (header) {
+  const links = [header.link, ...header.children.map(child => child.link)];
+  return links.some(link => link === route.hash);
+}
+
+function isItemActive (item) {
+  return item.link === route.hash;
+}
+
+watch(() => props.headers, (headers) => {
+  internalHeaders.value = headers;
+}, { immediate: true });
+
+watch(route, () => {
+  if (props.headers) return;
+
+  internalHeaders.value = usePageData().value.headers;
+}, { flush: 'pre', immediate: true, deep: true });
 </script>
 
-<style>
+<style lang="less" scoped>
 .dialtone-toc {
   width: var(--dt-size-850);
-  height: calc(100vh - var(--dt-size-700));
+    height: calc(100vh - var(--dt-size-700));
+    top: var(--dt-space-700);
+    right: 3%; // Magic number?
```

Comment:
Top and right should not be needed for the Page TOC if it's part of the grid, I only needed to add those for the tokens page, since the TOP was mounted from the tokens page, rather than from `Layout.vue`. 
And the magic numbers where the ones that I found that worked for best interaction with the token table.

Reply:
Do we need a prop to pass into this component to set these?

Reply:
I think it behaves correctly on both cases, can you see any issue by leaving it that way?

Reply:
I think this is a good addition to the components/utilities page too, looks cleaner and draws the user attention when clicking on a category.

Would be great to have some comments from @francisrupert @fede-dp here.

Reply:
For large monitors at least, the TOC looks really far away in pages other than the tokens page:

<img width="1908" alt="image" src="https://github.com/dialpad/dialtone/assets/24460973/b2388748-d8f9-4c63-b2dc-4bd3e7bba316">

What we have now, the TOC is on the grid:

<img width="1913" alt="image" src="https://github.com/dialpad/dialtone/assets/24460973/1e07f4fd-e3ec-4537-959d-f5f1db34e1f9">

Also, with a single glance I can see what contents are on the page.

Reply:
> I think this is a good addition to the components/utilities page too, looks cleaner and draws the user attention when clicking on a category.

Actually, I like it! Nice happy accident simplifying the TOC, and it even expands/collapses as you scroll.

And yeah, TOC positioning should be consistent among all pages. I didn't realize https://dialtone.dialpad.com/tokens/ is currently the only page that doesn't retain it within the grid. Take a stab at keeping keeping TOC position (does it even need positioning?). If it takes too long to resolve consistently, we can punt on that as a new task to figure out in the future.